### ISSUE TITLE
Lost everything

### ISSUE THREAD

**Issue by Willuscus** [TeamNewPipe/NewPipe#3960]
Hello,
With this new version, I lost everything in the app.
Every playlist, channel i was sub'd to, and settings in the app and I can't play any video, on YT source or Soundcloud and neither the other sources of content.

Wha' happened?
What Can I do ?

**Comment by InconsolableCellist** [comment]
Sadly you're going to have to get used to this, in my experience. I've lost all my data a few times now, and sometimes intentionally clearing all data in Android's app settings is the only way I can get the Guru Meditations to go away.

**Comment by FireMasterK** [comment]
> Sadly you're going to have to get used to this, in my experience. I've lost all my data a few times now, and sometimes intentionally clearing all data in Android's app settings is the only way I can get the Guru Meditations to go away.

You should really report the error instead.

**Comment by aggoi1234** [comment]
Make sure to export before resetting.

**Comment by wb9688** [comment]
Did you uninstall the previous version?

**Comment by vaelund** [comment]
I can confirm this bug. Same thing happened to me.
The probable chain of events:
* NewPipe informs me of a new Update (0.19.7), But the download links me to 0.19.6
* I just download and install 0.19.6 over my existing installation(i do not remember if that even did anything)
* After a reboot I have the same issues as Willuscus:
  * No saved playlists
  * No subscriptions (and importing subs gives me the error: File/Source does not exist)
  * Trying to play any video gives me an error

* I have since installed 0.19.7 and playing videos gives me the following error:

```
## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=ZglX4P-tpcA
* __Content Country:__ DE
* __Content Language:__ de-DE
* __App Language:__ de_DE
* __Service:__ YouTube
* __Version:__ 0.19.7
* __OS:__ Linux Android 9 - 28
<details><summary><b>Crash log </b></summary><p>

```
org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor$DecryptException: Could not parse decrypt function 
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.loadDecryptionCode(YoutubeStreamExtractor.java:786)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:670)
	at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:56)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:68)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:116)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$5fJcha6Sq5APJBLdG6osaJby-mc.call(Unknown Source:4)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.Maybe.subscribe(Maybe.java:4290)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.Flowable.subscribe(Flowable.java:14935)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.Maybe.subscribe(Maybe.java:4290)
	at io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: org.schabi.newpipe.extractor.utils.Parser$RegexException: failed to find pattern ";([A-Za-z0-9_\$]{2})\...\( inside of var iea=function(a){a=a.split("");var b=[-1215677809,null,-1744521562,833462946,694270393,function(c,d){d=(d%c.length+c.length)%c.length;c.splice(d,1)};"
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup(Parser.java:74)
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup(Parser.java:61)
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup1(Parser.java:52)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.loadDecryptionCode(YoutubeStreamExtractor.java:773)
	... 30 more

```
</details>
<hr>
 
```

**Comment by grravity** [comment]
Auto backups feature would "solve" this issue.

**Comment by Symbian9** [comment]
> Make sure to export before resetting.
>
> Auto backups feature would "solve" this issue.

Just export database before each version upgrade is the easiest way to backup.

**Comment by wb9688** [comment]
@Willuscus and @vaelund: What phone do you have? (And did you uninstall NewPipe before installing the new version?)

**Comment by vaelund** [comment]
Blackview A60Pro
No, I did not uninstall the old version.

**Comment by wb9688** [comment]
@vaelund: That's strange then, what version of NewPipe did you previously (before updating to v0.19.8) have installed?

**Comment by vaelund** [comment]
0.19.6, please also check my description in my first comment(this is as much as I remember, sry).

**Comment by Gaffen** [comment]
Just encountered this on 0.20.0; installed the update directly from the git repo after the app prompted me to (wasn't available on f-droid at the time).

My device's battery is aging and ran out of juice whilst using the application. On restart all playlists and subscriptions were missing - as if I had just started a fresh install.

AntennaPod has had a similar issue and implemented an autodump feature so that an emergency backup was available in these situations. Is this something that is being considered/could be implemented for NewPipe?

It's a shame to lose everything, is there any kind of warning in the app at the moment about this behaviour on update? It might be an idea to warn users about this potential behaviour if it is quite common, at least until a more stable update pattern is established.

**Comment by opusforlife2** [comment]
At least make sure you export database once in a while in case problems like this occur. There is no autodump feature, but there is a PR open for Auto backup of the app.

**Comment by Gaffen** [comment]
Yeah, I've learned that lesson now, lol. I guess I was suggesting at least a warning for the uninformed; if there was one I missed it.

Either way, in spite of this I really am enjoying this app. If I find some spare time I hope to dedicate some of it to squashing some bugs. I'll take a look for that pull request!

**Comment by shin-bot87** [comment]
Try to sinc with your YT account throught Micro G for New Pipe. Now you can't do nothing, but maybe the next time if you have everything sinc with your account, you can restore it if it happens again.

**Comment by litetex** [comment]
TL;DR 
Make backups regularly - this is applies to everything that can hold data and not only NewPipe.

Closing due to inactivity.
